packages/media/src/test/getFileExtFromUrl.js

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * Internal dependencies
+ */
+import getFileExtFromUrl from '../getFileExtFromUrl';
+
+describe('getFileExtFromUrl', () => {
+  it.each([
+    'https://www.example.com/test.jpg',
+    'https://www.example.com/path/test.jpg',
+    'test.jpg',
+  ])('should %s find ext', (url) => {
+    expect(getFileExtFromUrl(url)).toStrictEqual('jpg');
+  });
+  it.each([
+    'https://www.example.com/test',
+    'https://www.example.com/',
+    '#',
+    '-1',
+    '',
+  ])('should %s not find ext', (url) => {
+    expect(getFileExtFromUrl(url)).toStrictEqual('');
+  });
+
+  it('should null not find ext', () => {
+    expect(getFileExtFromUrl(null)).toBeNull();
+  });
```

Comment:
Sorry for not being clear here. `it.each` could be used for _all_ assertions, passing input & output.

Example:

```suggestion
  it.each([
    ['https://www.example.com/test.jpg', 'jpg'],
    ['https://www.example.com/path/test.jpg', 'jpg'],
    ['test.jpg', 'jpg'],
    ['https://www.example.com/test', ''],
    ['https://www.example.com/', ''],
    ['#', ''],
    ['-1', ''],
    ['', ''],
    [null, null],
  ])('should get file extension from URL %s', (url, expectedOutput) => {
    expect(getFileExtFromUrl(url)).toStrictEqual(expectedOutput);
  });
```

Reply:
I know, but the title of the test isn't super clear. Breaking the tests up makes for cleaner test titles.

Reply:
The current titles aren't super clear either.

The spec title can easily be reworded, that should be the least problem. There are even many formatting options for that: https://jestjs.io/docs/api#testeachtablename-fn-timeout

A data provider like this makes it way easier to add more test cases.

I don't think it can get any more readable than this:

```suggestion
  it.each`
    url                                        | expected
    ${'https://www.example.com/test.jpg'}      | ${'jpg'}
    ${'https://www.example.com/path/test.jpg'} | ${'jpg'}
    ${'test.jpg'}                              | ${'jpg'}
    ${'https://www.example.com/test'}          | ${''}
    ${'https://www.example.com/'}              | ${''}
    ${'#'}                                     | ${''}
    ${'-1'}                                    | ${''}
    ${''}                                      | ${''}
    ${null}                                    | ${undefined}
  `('should return "$expected" for url "$url"', ({ url, expected }) => {
    expect(getFileExtFromUrl(url)).toStrictEqual(expected);
  });
```